### ISSUE TITLE
Accessibility and Diversity best-practices

### ISSUE THREAD

**Issue by aschrijver** [gabek/fedidocs#14]
**TL;DR**: The level of attention to Accessibility and Diversity is a USP of the Fediverse. Document best-practices for related features to be present in federated apps, so that the Fediverse remains a shining example of "social networking done right".

---

On the Fediverse there has always been a bigger focus on applying 'humane technologies', and the development of these went hand in hand with the social culture and dynamics that distinguishes the Fediverse from its corporate counterparts (more focused on growth, engagement, surveillance capitalism and ads, "user is the product", etc.)

This led to implementing social feature differently, more thoughtfully.. and to best-practices for adoption in fedi apps.

There are multiple examples of this different approach, like [Anti-Search](https://www.tbray.org/ongoing/When/202x/2022/12/30/Mastodon-Privacy-and-Search#p-2) described by @timbray, and in particular the practice to add captions to images. App support on the latter is only recently improving with features that make people aware when captions aren't present. Yet still many folks forget to add them, and need to be reminded by others. There are even bots that caption or give heads-up to other fedizens to reply with a caption.

There's a sizable group of vision impaired or blind people that use fedi apps, but also many fedizens that are neurodivergent e.g. having ADD/ADHD or having issues on the autism spectrum. On the latter for instance a [recent `fediverse-idea` was posted](https://codeberg.org/fediverse/fediverse-ideas/issues/20) to standardize a set of custom emoji's related to A11Y.

As part of giving the best UX for any group in society to join the Fediverse the various best-practices and examples of how they are implemented in apps can be collected and documented as best-practices in FediDocs so they aren't forgotten by newcomer devs.

**Comment by benpate** [comment]
I'm 100% for Accessibility and Inclusion in all things tech (despite my minimal experience in this area).  Do you have a high-level list of the issues to address?  For example:

- Optimizing UX for screen readers
- Making (requiring?) captions for images
- Inclusive emojis and shortcodes (from your link)
- ???

I'm sure there are tons of things that I'm missing.  It could be very valuable to generate a good, workable list here.

**Comment by aschrijver** [comment]
I don't have such list, but just as in the chatroom discussion there's mention of collecting User Stories, such list may be formed. Your bullets are the start :)

An accessibility track to have this be a theme within these docs, and where many fedizens with disabilities can be eager contributors of user feedback.

**Comment by deborahgu** [comment]
Obviously you're planning on starting with WCAG and the WAI introductory docs, right? There is no reason to reinvent the wheel on this one; WAI even has [user stories](https://www.w3.org/WAI/people-use-web/user-stories/) that absolutely apply to the fediverse. And, for anything that involves user created content, it should follow [ATAG] (https://www.w3.org/WAI/standards-guidelines/atag/) as well, even though those guidelines are older.

Best practice can be fine tuned later, but there's no reason to reinvent the wheel.

* the WAI user stories and personas are already great
* the minimal guideline should be aim for WCAG A compliance in all applications
* user-created content sections should conform to ATAG

and then fine-tune later. Making the content perceivable (by low vision folks, colorblind folks, screen reader and braille reader users, and d/Deaf and hard of hearing users) and operable (by  blind, keyboard-only, speech-to-text, tablet, and switch users) is important, and how to do that is very well specified already.

**Comment by gabek** [comment]
I'm also 100% for accessibility and Inclusive development. But I do think discussing images and emoji and any kind of interface is a bit out of scope when it comes to Fediverse interop.

**Comment by aschrijver** [comment]
I'd say.. depends. First of all there's a number of apps that support "reactions", and there's mention of `EmojiReact`, which is AP-related. Then, when in general the "UX of the Fediverse" is also in scope of FediDevs to give attention, it doesn't make sense to focus on UX of individual apps (imho), but there's UX that relates to the unique culture and 'fediquette' that exists and has matured for years. This is something that is a 'unique selling point' and to be retained. A recommendation for a "common emoji set" to be standardized may well fit that.
